bigtable-hbase-mirroring-client-2.x-parent/bigtable-hbase-mirroring-client-2.x/src/main/java/com/google/cloud/bigtable/mirroring/hbase2_x/utils/AsyncRequestScheduling.java

```diff
@@ -78,6 +93,40 @@
           } finally {
             reservation.release();
           }
+          return null;
         });
   }
+
+  /**
+   * Class containing an object of interest to user and a future which completes when background
+   * actions related to it are completed.
```

Comment:
I'm sorry, but the comment isn't clear to me.

How about this:
> Objects of this class gather futures which are completed on different stages of operation execution.
> 
> These objects are used internally to register continuations to those events (e.g. complete the future handed to the user or release resources after verification is done).
>
> @param <T> the type representing the stage of notifying the user; usually it's simply a `CompletableFuture<U>`, but in case of batch requests it is a `List<CompletableFuture<U>>`